client/canary/scripts/gitopscluster_test.sh

```diff
@@ -0,0 +1,365 @@
+
+# Create managedclusterset
+$KUBECTL_HUB apply -f scripts/argocd/managedclusterset.yaml
+echo "$(date) managedclusterset created"
+
+# Add all managed clusters to managedclusterset clusterset1
+MANAGED_CLUSTERS=( $($KUBECTL_HUB get managedclusters -o name |awk -F/ '{print $2}') )
+
+for element in "${MANAGED_CLUSTERS[@]}"
+do
+   echo "$(date) Adding ${element} to managed cluster set clusterset1"
+   $KUBECTL_HUB label managedclusters ${element} cluster.open-cluster-management.io/clusterset=clusterset1
+done
+
+# Create ManagedClusterSetBinding
+$KUBECTL_HUB apply -f scripts/argocd/managedclustersetbinding.yaml
+echo "$(date) managedclustersetbinding created"
+
+# Create placement to choose all managed clusters
+sed -i -e "s/__NUM__/${#MANAGED_CLUSTERS[@]}/" scripts/argocd/placement.yaml
+if [ $? -ne 0 ]; then
+    echo "$(date) failed to substitue __NUM__ in placement.yaml"
+    echo "E2E CANARY TEST - EXIT WITH ERROR"
+    exit 1
+fi
+$KUBECTL_HUB apply -f scripts/argocd/placement.yaml
+echo "$(date) placement created"
+
+# Sleep for placement decision
+sleep 10
+
+# Install ArgoCD CLI
+rm -fr /usr/local/bin/argocd
+
+if [[ "$LOCAL_OS" == "Linux" ]]; then
+    curl -sSL -o /usr/local/bin/argocd https://github.com/argoproj/argo-cd/releases/download/$ARGO_VERSION/argocd-linux-amd64
+elif [[ "$LOCAL_OS" == "Darwin" ]]; then
+    curl -sSL -o /usr/local/bin/argocd https://github.com/argoproj/argo-cd/releases/download/$ARGO_VERSION/argocd-darwin-amd64
+fi
+
+chmod +x /usr/local/bin/argocd
+
+# Create GitOpsCluster for argocdtest1
+$KUBECTL_HUB delete -f scripts/argocd/gitopscluster.yaml
+$KUBECTL_HUB apply -f scripts/argocd/gitopscluster.yaml
+echo "$(date) gitopscluster created"
+
+# Sleep for GitOpsCluster reconcile
+sleep 10
+
+verifyClusterRegistrationInArgo "argocdtest1"
+
+# Change the target the second ArgoCD instance in argocdtest2
+$KUBECTL_HUB -n default patch gitopscluster gitops-cluster-test -p '{"spec": {"argoServer": {"argoNamespace": "argocdtest2"}}}' --type merge
+
+# Sleep for GitOpsCluster reconcile
+sleep 10
+
+# Verify that the managed cluster secrets are deleted from the first argocd instance
+echo "$(date)  ====  verify that the managed cluster secrets are deleted from the first argocd instance"
+for element in "${MANAGED_CLUSTERS[@]}"
+do
+   verifySecretDeleted ${element} "argocdtest1"
+done
+
+verifyClusterRegistrationInArgo "argocdtest2"
+
+# Change the target the default ArgoCD instance in openshift-gitops operator
+$KUBECTL_HUB -n default patch gitopscluster gitops-cluster-test -p '{"spec": {"argoServer": {"argoNamespace": "openshift-gitops"}}}' --type merge
+
+# Sleep for GitOpsCluster reconcile
+sleep 10
+
+# Verify that the managed cluster secrets are deleted from the second argocd instance
+echo "$(date)  ====  verify that the managed cluster secrets are deleted from the second argocd instance"
+for element in "${MANAGED_CLUSTERS[@]}"
+do
+   verifySecretDeleted ${element} "argocdtest"
```

Comment:
@rokej  It seems we should verify all secrets are removed from `argocdtest2`  as the argoNamespace has been updated from `argocdtest2` to `open-shift-gitops` in the gitopscluster object
```
verifySecretDeleted ${element} "argocdtest"
```

Reply:
@xiangjingli You are right. Thanks! Another PR coming for this change.